#! /bin/sh /usr/share/dpatch/dpatch-run
## 99-unnamed.dpatch by  <ch@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Bugfix für MySQL#21011

@DPATCH@

--- old/client/mysql_upgrade.c	2006-08-08 01:09:23.000000000 +0200
+++ new/client/mysql_upgrade.c	2006-08-16 00:38:42.000000000 +0200
@@ -180,7 +192,7 @@
   }
   buffer_end= strnmov(buffer, "\n[client]", sizeof(buffer));
   if (opt_password)
-    buffer_end= strxnmov(buffer, sizeof(buffer),
+    buffer_end= strxnmov(buffer_end, sizeof(buffer),
                          "\npassword=", opt_password, NullS);
   error= my_write(defaults_file, buffer, (int) (buffer_end - buffer),
                   MYF(MY_WME | MY_FNABP));
